### DISCUSSION TITLE
How to set object' orientations?

### DISCUSSION THREAD

**Discussion by dexin-wang** [StanfordVL/BEHAVIOR-1K#1050] [General]
I noticed that the OnTop and Inside functions sample the object rotation based on the object's orientations, and orientations are all empty lists.
I want to sample the x or y axis rotation when setting the OnTop and Inside states of the object. Is this possible?
I currently think of an indirect method, which is to rotate the object along the x and y axes after setting the OnTop state, and adjust the object's z coordinate based on the aabb before and after.
But is there a more direct way?
Thank you very much.

**Comment by hang-yin** [comment]
@ChengshuLi probably has a great answer to this.

**Comment by ChengshuLi** [comment]
@dexin-wang For the OnTo and Inside sampling, the current code already randomizes orientation in the z-axis (yaw angle). And the code also assumes that the default orientation ([0, 0, 0, 1]) is a stable orientation. 


If you want to randomize the orientation more aggressively, you can sample a random orientation in the following lines:
https://github.com/StanfordVL/OmniGibson/blob/255a4e4fb42fa4d2f1a43a94de37085474db4377/omnigibson/utils/object_state_utils.py#L134-L137

The downside is that this random orientation might not be stable, so even if you successfully place the object, it might roll away.
